pkg/domain/infra/abi/play.go

```diff
@@ -756,6 +756,15 @@ func (ic *ContainerEngine) playKubePod(ctx context.Context, podName string, podY
 					return nil, nil, err
 				}
 			}
+		} else if v.Type == kube.KubeVolumeTypeImage {
+			imageName := v.Source
+			_, fullName, err := ic.Libpod.LibimageRuntime().LookupImage(v.Source, nil)
+			if err != nil {
+				return nil, nil, fmt.Errorf("image %s, used as pod volume, does not exist in local storage: %w", imageName, err)
+			}
```

Comment:
Note that https://github.com/containers/podman/issues/23775 seems to agree that kube play should pull the image otherwise users have to know the image in advance and cannot just play a "random" yaml.

Also you never validate the pull policy which means user could set anything there and given k8s defaults to always this is rather confusing that you throw an error here

Reply:
Yes. That's not ideal. In the documentation I have mentioned that images need to be pre-pulled. That's consistent with the existing support for automounting images through annotations. I have made the simplest choice for this PR but we should decide (for this PR or in a separate one):
- live with an inconsistencies of these two image mounting techniques
- implement pre-pull for both
- remove support for image automounting through annotations

Reply:
Yes, this is why I wanted to complete the discussion on #23775 before merging this change.
Please note that we can't remove the annotation support as this will be an API break. The problem with automatic pulling for the annotation case is the PullPolicy. Users will expect images to not be updated while if implemented, can happen.
So, I think the right path is to deprecate the annotations option (without removing it) and live with the inconsistency for the time being